src/metabase/lib/temporal_bucket/util.cljc

```diff
@@ -47,9 +54,39 @@
     next-interval-message
     next-single-interval-message
     next-plural-interval-message]
-   `(cond
-      (zero? ~n) (i18n/tru ~this-interval-message)
-      (= ~n -1)  (i18n/tru ~prev-interval-message)
-      (= ~n 1)   (i18n/tru ~next-interval-message)
-      (neg? ~n)  (i18n/trun ~prev-single-interval-message ~prev-plural-interval-message (abs ~n))
-      (pos? ~n)  (i18n/trun ~next-single-interval-message ~next-plural-interval-message ~n))))
+   (assert (every? string? [this-interval-message
+                            prev-interval-message
+                            prev-single-interval-message
+                            prev-plural-interval-message
+                            next-interval-message
+                            next-single-interval-message
+                            next-plural-interval-message])
+           "[[temporal-interval-tru]] accepts compile-time strings only.")
+   `(let [n# ~n]
+      (cond
+        (zero? n#) (i18n/tru ~this-interval-message)
+        (= n# -1)  (i18n/tru ~prev-interval-message)
+        (= n# 1)   (i18n/tru ~next-interval-message)
+        (neg? n#)  (i18n/trun ~prev-single-interval-message ~prev-plural-interval-message (abs n#))
+        (pos? n#)  (i18n/trun ~next-single-interval-message ~next-plural-interval-message n#)))))
```

Comment:
When would one want to provide `prev-single-interval-message` and `next-single-interval-message`?

We might want to throw an exception on the else branch rather than returning `nil`.

Reply:
@metamben fixed